bin/alluxio-start.sh

```diff
@@ -47,14 +47,36 @@ get_env() {
   . ${ALLUXIO_LIBEXEC_DIR}/alluxio-config.sh
 }
 
+# The exit status is 0 if ALLUXIO_RAM_FOLDER is mounted as tmpfs or ramfs.
+is_ram_folder_mounted() {
+  if [[ -z ${ALLUXIO_RAM_FOLDER} ]]; then
+    return 1
+  fi
+  local mounted_fs=""
+  if [[ $(uname -s) = Darwin ]]; then
+    mounted_fs=$(mount -t "hfs" | grep '/Volumes/' | cut -d " " -f 3)
+  else
+    mounted_fs=$(mount -t "tmpfs,ramfs" | cut -d " " -f 3)
+  fi
+
+  for fs in ${mounted_fs}; do
+    if [[ "${ALLUXIO_RAM_FOLDER}" = "${fs}" || \
+     "${ALLUXIO_RAM_FOLDER}" =~ ^"${fs}"\/.* ]]; then
+      return 0
+    fi
+  done
+
+  return 1
+}
+
 check_mount_mode() {
   case "$1" in
     Mount);;
     SudoMount);;
     NoMount)
-      mount | grep "${ALLUXIO_RAM_FOLDER}" > /dev/null
-      if [[ $? -ne 0 || -z ${ALLUXIO_RAM_FOLDER} ]]; then
-        if [[ $(uname -s) == Darwin ]]; then
+      is_ram_folder_mounted
+      if [[ $? -ne 0 ]]; then
+        if [[ $(uname -s) = Darwin ]]; then
```

Comment:
I don't know which is preferred in Alluxio. I've seen both of them used.

As i found, == behaves differently in [[]] and []. http://www.tldp.org/LDP/abs/html/comparison-ops.html

I changed it back to == since it seems that you prefer ==?